### PR TITLE
Overhall to docs adding FAQ section

### PR DESCRIPTION
**Pull request by thchang**
Overhall to docs:
 - updated quickstart example and other examples to use recommended inputs
 - add a FAQ document with additional details
 - added a tutorial on high-dimensional / local model-based optimization on a limited budget
 - added documentation for and debugged some minor issues in "format" specifier for all output methods
 - added format specifier to libE_MOOP class
 - changed several examples to use pandas dataframes, which significantly improves the legibility of example outputs

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -314,13 +320,15 @@ Our online documentation:
 .. _Actions: https://github.com/parmoo/parmoo/actions
 .. _CONTRIBUTING: https://github.com/parmoo/parmoo/blob/main/CONTRIBUTING.rst
 .. _dash: https://dash.plotly.com
+.. _FAQs: https://parmoo.readthedocs.io/en/latest/faqs.html
```
Comment (wildsm) [inline README.rst:323]: This url will be generated after merge, correct?

Comment (thchang) [reply README.rst:323]: yes, that will be the url of the faq section once we merge to main

**Comment by thchang** [comment]
It looks like something is going on with libE latest version -- I just upgraded the version on my local machine and I am now getting the same error as the GH servers -- it is an error on import, so I think it's a system-level issue not a ParMOO issue but I don't know how to fix it yet

**Comment by thchang** [comment]
I just talked to libE team and it is the latest release of pydantic (ver 2.0, released Jun 30), which appears to have some API-breaking changes for libEnsemble

**Review by wildsm — APPROVED** [review]
Good, with the exception of the possible `how-to-write.rst`

File: docs/how-to-write.rst [hunk 2]
```diff
@@ -796,6 +796,14 @@ After solving the MOOP, you can view the results using
 
     soln = moop.getPF()
 
+The output format defaults to a numpy structured array.
+However, you can change it to a pandas dataframe using the optional
+``format`` argument.
```
[the remark this answers is not in the document]
Comment (thchang) [reply docs/how-to-write.rst:801]: Yes, thanks for reminding

[the remark this answers is not in the document]
Comment (thchang) [reply docs/how-to-write.rst:801]: @wildsm -- see latest commit

**Pull request merged**